### PR TITLE
Fix2741

### LINKED ISSUE #2741: Power8 - Monerod not syncing
`2017-10-30 17:17:16.156     100064abf170        WARN    blockchain.db.lmdb      src/blockchain_db/lmdb/db_lmdb.cpp:72   Failed to create a read transaction for the db: MDB_READERS_FULL: Environment maxreaders limit reached
2017-10-30 17:17:16.157     100061dbf170        WARN    blockchain.db.lmdb      src/blockchain_db/lmdb/db_lmdb.cpp:72   Failed to create a read transaction for the db: MDB_READERS_FULL: Environment maxreaders limit reached
2017-10-30 17:17:16.162     100066dbf170        WARN    blockchain.db.lmdb      src/blockchain_db/lmdb/db_lmdb.cpp:72   Failed to create a read transaction for the db: MDB_READERS_FULL: Environment maxreaders limit reached
2017-10-30 17:17:16.219     1000159bf170        WARN    blockchain.db.lmdb      src/blockchain_db/lmdb/db_lmdb.cpp:72   Failed to create a read transaction for the db: MDB_READERS_FULL: Environment maxreaders limit reached
2017-10-30 17:17:16.219     100014fbf170        WARN    blockchain.db.lmdb      src/blockchain_db/lmdb/db_lmdb.cpp:72   Failed to create a read transaction for the db: MDB_READERS_FULL: Environment maxreaders limit reached
2017-10-30 17:17:27.162 [P2P3]  WARN    blockchain.db.lmdb      src/blockchain_db/lmdb/db_lmdb.cpp:72   Failed to renew a read transaction for the db: Invalid argument
2017-10-30 17:17:27.373 [P2P7]  WARN    blockchain.db.lmdb      src/blockchain_db/lmdb/db_lmdb.cpp:72   Failed to renew a read transaction for the db: Invalid argument
2017-10-30 17:17:27.967 [P2P5]  WARN    blockchain.db.lmdb      src/blockchain_db/lmdb/db_lmdb.cpp:72   Failed to renew a read transaction for the db: Invalid argument
2017-10-30 17:17:30.184 [P2P5]  WARN    blockchain.db.lmdb      src/blockchain_db/lmdb/db_lmdb.cpp:72   Failed to renew a read transaction for the db: Invalid argument
`
Ubuntu 16.04

### PR DESCRIPTION
**Pull request by hyc**
Don't overallocate threads, set LMDB maxreaders if larger number of threads are in use. Fixes #2741

### CODE REVIEW

File: src/common/threadpool.cpp [hunk 1]
```diff
@@ -74,7 +74,7 @@ void threadpool::submit(waiter *obj, std::function<void()> f) {
 }
 
 int threadpool::get_max_concurrency() {
-  return max / 2;
+  return tools::get_max_concurrency();
```
Comment (vtnerd) [inline src/common/threadpool.cpp]: Should this be `return max;` ?

Comment (hyc) [reply src/common/threadpool.cpp]: Ah sure, that'd be better. Thanks.

**Comment by moneromooo-monero** [comment]
Can you squash the "tweak" commit please ?

**Comment by hyc** [comment]
Squashed

**Pull request merged**